src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -51,6 +51,12 @@ private sealed class State
             public readonly TaskCompletionSource<uint> ConnectTcs = new TaskCompletionSource<uint>(TaskCreationOptions.RunContinuationsAsynchronously);
             public readonly TaskCompletionSource<uint> ShutdownTcs = new TaskCompletionSource<uint>(TaskCreationOptions.RunContinuationsAsynchronously);
 
+            // Note that there's no such thing as ressetable TCS, so we need reallocate the TCS every time we set the result.
+            // We also cannot use solutions like ManualResetValueTaskSourceCore, since we can have multiple waiters on the same TCS.
+            // To avoid as much as possible allocations, we keep the TCSes null until someone explicitely asks for them in WaitForAvailableStreamsAsync.
```

Comment:
"keep the TCSes null" is a bit awkward, I would suggest "allocate when needed"

Reply:
Fixed, reworded.